include/seqan3/search/all.hpp

```diff
@@ -35,9 +35,56 @@
  * SeqAn currently supports very fast FM indices. For more details visit the \ref submodule_fm_index
  * "FM Index submodule".
  *
+ * ## Search Algorithms
+ *
+ * The Search module offers a simple unified interface that allows searching FM indices and choosing the best
+ * algorithm based on the index at hand.
+ *
+ * ## FM Indices
+ *
+ * The search algorithms for FM indices implement either a trivial backtracking approach or an optimum search scheme.
+ * Latter are currently only available for searches with up to and including three errors using bidirectional indices.
+ * The optimum search schemes will be improved in the future to handle unidirectional indices and higher error counts.
+ *
+ * \if DEV
+ *
+ * ### Implementation details of Search Schemes
+ *
+ * A search scheme is a collection of searches, where each search `s` specifies the order in which the blocks are
+ * searched (`s.pi`), the lower error bounds (`s.l`) and the upper error bounds (`s.u`). If the number of blocks that
+ * the query sequences are split into are known at compile time, the data structure `search` is recommended, otherwise
```

Comment:
```suggestion
 * the query sequences are split into are known at compile time, the data structure seqan3::detail::search is recommended, otherwise
```